Userland/Libraries/LibWeb/HTML/HTMLOptionsCollection.cpp

```diff
@@ -71,6 +73,52 @@ WebIDL::ExceptionOr<void> HTMLOptionsCollection::set_length(WebIDL::UnsignedLong
     return {};
 }
 
+// https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmloptionscollection-setter
+WebIDL::ExceptionOr<void> HTMLOptionsCollection::set_value_of_indexed_property(u32 index, JS::Value unconverted_option)
+{
+    // The spec doesn't seem to require this, but it's consistent with length handling and other browsers
+    if (index > 100'000) {
+        return {};
+    }
+
+    // 1. If value is null, invoke the steps for the remove method with index as the argument, and return.
+    if (unconverted_option.is_null()) {
+        remove(static_cast<WebIDL::Long>(index));
+        return {};
+    }
+
+    if (!unconverted_option.is_object() || !is<HTMLOptionElement>(unconverted_option.as_object())) {
+        return WebIDL::TypeMismatchError::create(realm(), "The value provided is not an HTMLOptionElement"_fly_string);
+    }
+
+    auto& option = dynamic_cast<HTMLOptionElement&>(unconverted_option.as_object());
```

Comment:
Should be okay to `static_cast<HTMLOptionElement&>` here as you have verified just above that it is indeed an option element.

Reply:
changed